Helper/Address.php

```diff
@@ -0,0 +1,120 @@
+    /**
+     * @param AddressAdapterInterface $address
+     * @param $houseNumberStreetLine
+     * @param $customerStreetLinesEnabled
+     * @return array
+     */
+    public function getStreetAndHouseNumberFromAddress(
+        AddressAdapterInterface $address,
+        $houseNumberStreetLine,
+        $customerStreetLinesEnabled
+    ): array {
+        $addressArray = [
+            $address->getStreetLine1(),
+            $address->getStreetLine2(),
+            $address->getStreetLine3(),
+            $address->getStreetLine4()
+        ];
+
+        // Cap the full street to the enabled street lines
+        $street = array_slice($addressArray, 0, $customerStreetLinesEnabled);
+
+        if ($this->isSeparateHouseNumberRequired($address->getCountryId())) {
+
+            $drawHouseNumberWithRegex = $houseNumberStreetLine == 0 ||
+                $houseNumberStreetLine > $customerStreetLinesEnabled;
+
+            if ($drawHouseNumberWithRegex) {
+                // House number street line is disabled or there aren't enough street lines, use the regex
+                return $this->getStreetAndHouseNumberFromArray($street);
+            } else {
+                // Extract and remove the house number from the street name array
+                $houseNumber = $street[$houseNumberStreetLine - 1];
+                unset($street[$houseNumberStreetLine - 1]);
+                return $this->formatAddressArray(implode(' ', $street), $houseNumber);
+            }
+        }
+        // Return the full street and house number = N/A
+        return $this->formatAddressArray(implode(' ', $street), 'N/A');
+    }
+
+    /**
+     * @param string[] $addressArray
+     * @return array
+     */
+    private function getStreetAndHouseNumberFromArray(array $addressArray): array
+    {
+        $addressString = implode(' ', $addressArray);
+
+        preg_match(
+            self::HOUSE_NUMBER_REGEX,
+            trim($addressString),
+            $houseNumber,
+            PREG_OFFSET_CAPTURE
+        );
+
+        if (!empty($houseNumber['0'])) {
+            $_houseNumber = trim($houseNumber['0']['0']);
+            $position = $houseNumber['0']['1'];
+            $streetName = trim(substr($addressString, 0, $position));
+            return $this->formatAddressArray($streetName, $_houseNumber);
+        }
+        return $this->formatAddressArray($addressString, 'N/A');
+    }
+
+    /**
+     * @param $street
+     * @param $houseNumber
+     * @return array
+     */
+    private function formatAddressArray($street, $houseNumber): array
+    {
+        return (['name' => trim($street), 'house_number' => trim($houseNumber)]);
+    }
+
+    /**
+     * Checks if the house number needs to be sent to the Adyen API separately or as it is in the street field
+     *
+     * @param $country
+     * @return bool
+     */
+    private function isSeparateHouseNumberRequired($country)
+    {
+        $countryList = ["dk", "fi", "uk", "us", "se", "at", "no"];
```

Comment:
separate house numbers are not necessarily required by country but also payment method. I think this list is only applicable for Klarna but for example afterpay require the houseNumberOrName filled with a correct data and not just N/A

I think the best approach would be to stop relying to this function and only use the new configuration that you created.

In case the config says use the street line 1-2-3-4 then simply use those as houseNumberOrName otherwise fall back to the regex and try to parse it.

Reply:
Sounds good! I'll remove this check then, and only rely on the config

Reply:
Done https://github.com/Adyen/adyen-magento2/pull/977/commits/b66a11fee9c716436419126a98aa994bf410e0a8